Tweaks to user orientaton on the getting started + edit of new(ish) trouble shooting (#666)
spatialos/gdk-for-unity: docs/content/troubleshooting.md

**Workaround**<br/>

To remove the error message, disable burst compilation by unchecking **Jobs** > **Enable Burst Compilation** in your Unity Editor.

<br/>
## Warnings
<br/>
#### <li> <b>I get several warnings when opening the project for the first time</b>

<br/>
**Cause**<br/>
These warnings originate from code written by Unity, for example; `com.unity.mathematics` or `com.unity.entities`. 

**Fix**<br/>
These warnings are benign so you can ignore them.

<br/>
#### <li> <b> I get several warnings when deleting a GameObject or ECS entity that represents a SpatialOS entity</b>

**Cause**<br/>
These warnings occur when a [GameObject]({{urlRoot}}/content/glossary#gameobject) or [ECS entity]({{urlRoot}}/content/glossary#unity-ecs-entity) that represents a [SpatialOS entity]({{urlRoot}}/content/glossary#spatialos-entity) is deleted before its corresponding SpatialOS entity leaves the [worker's view]({{urlRoot}}/content/glossary#worker-s-view). 

> The GDK expects a GameObject and ECS entity that represent a SpatialOS entity are only deleted after the SpatialOS entity they represent has left that worker's view: 

Reviewer: I was going to change "ECS entity" to "ECS Entity", as I thought we started out by capitalising it in the GDK docs. However, looking in the GDK docs now, it's inconsistent. I seem to recall it was also inconsistent in Unity's basic ECS docs, in which case we can probably just make a call one way or the other.

Author: I think we ended up going with "entity" lower case. (A fair few of the Unity docs didn't get an editorial pass due to time panic so are likely to have inconsistencies - also I am not always on PRs.)

> The GDK expects a GameObject and ECS entity that represent a SpatialOS entity are only deleted after the SpatialOS entity they represent has left that worker's view: 

Reviewer: ```suggestion
The GDK expects that a GameObject and ECS entity that represent a SpatialOS entity are only deleted after the SpatialOS entity they represent has left that worker's view: 
```

- added "that" to set up the sentence structure (I had originally assumed it would go on to say something like "The GDK expects a GameObject to do XYZ")
- this talks about "a GameObject and ECS entity" and is in the plural, whereas line 88 says "a GameObject or ECS entity" and uses the singular. Is one of these wrong?

> The GDK expects that a GameObject and ECS entity that represent a SpatialOS entity are only deleted after the SpatialOS entity they represent has left that worker's view: 

Author: ```suggestion
 The GDK expects that a GameObject or ECS entity that represents a SpatialOS entity is only deleted after the SpatialOS entity it represents has left that worker's view: 
```

> * An ECS entity is deleted automatically by the GDK when its corresponding SpatialOS entity leaves that worker's view. 

Reviewer: ```suggestion
* The GDK automatically deletes an ECS entity when its corresponding SpatialOS entity leaves that worker's view. 
```

> * For GameObjects, you need to delete the GameObject via your code. The SpatialOS [Runtime]({{urlRoot}}/content/glossary#spatialos-runtime) sends a message to the server-worker when a SpatialOS entity leaves that worker's view, the GDK passes this message on to you; it's your responsibility to act upon these commands at the right time in your code.

Reviewer: ```suggestion
* For GameObjects, you need to delete the GameObject via your code. The SpatialOS [Runtime]({{urlRoot}}/content/glossary#spatialos-runtime) sends a message to the server-worker when a SpatialOS entity leaves that worker's view, and the GDK passes this message on to you; it's your responsibility to act upon these commands at the right time in your code.
```

  * Ensure that you only access Requirables in a MonoBehaviour while the MonoBehaviour is enabled. Requirables are supposed to be null while the MonoBehaviour is disabled. Note that certain Unity event methods like `OnTriggerEnter` or `OnCollisionEnter` may be invoked, even if a MonoBehaviour is disabled.
  [comment]: <> (TODO: make this fix clearer - what is meant by "Requireables" - we haven't defined this. JIRA: https://improbableio.atlassian.net/browse/UTY-1575 and https://improbableio.atlassian.net/browse/TC-169.)

<br/>
#### <li> <b> A MonoBehaviour containing fields using the `Require` attribute remains disabled when I expect it to be enabled </b>

**Causes and fixes**<br/>

This can be caused by multiple problems. Follow the checklist below to discover the cause and fix:

  * Ensure that `GameObjectRepresentationHelper.AddSystems` is called when initializing your [worker]({{urlRoot}}/content/glossary#worker).

>   * Ensure that the GameObject containing your MonoBehaviour is associated with a [SpatialOS entity]({{urlRoot}}/content/glossary#spatialos-entity). You can verify this be examining whether a `SpatialOSComponent` MonoBehaviour was added to your GameObject at [Runtime]({{urlRoot}}/content/glossary#spatialos-runtime).

Reviewer: I don't think this should link to "SpatialOS Runtime", since "runtime" here is being used in the sense of "while running".

Author: Mmm - I see. Think we should use a different way to say this completely.

Author: "   * Ensure that the GameObject containing your MonoBehaviour is associated with a [SpatialOS entity]({{urlRoot}}/content/glossary#spatialos-entity). You can verify this by examining whether a `SpatialOSComponent` MonoBehaviour was added to your GameObject when you run your game.

Author: @anne-edwards Would you make the suggested change so I can add it? I can't see to add my own edits.

>   * Ensure that the GameObject containing your MonoBehaviour is associated with a [SpatialOS entity]({{urlRoot}}/content/glossary#spatialos-entity). You can verify this be examining whether a `SpatialOSComponent` MonoBehaviour was added to your GameObject at [Runtime]({{urlRoot}}/content/glossary#spatialos-runtime).

Author: ```suggestion
   * Ensure that the GameObject containing your MonoBehaviour is associated with a [SpatialOS entity]({{urlRoot}}/content/glossary#spatialos-entity). You can verify this be examining whether a `SpatialOSComponent` MonoBehaviour was added to your GameObject when you run your game.
```

>   * Ensure that the GameObject containing your MonoBehaviour is associated with a [SpatialOS entity]({{urlRoot}}/content/glossary#spatialos-entity). You can verify this be examining whether a `SpatialOSComponent` MonoBehaviour was added to your GameObject at [Runtime]({{urlRoot}}/content/glossary#spatialos-runtime).

Author: ```suggestion
   * Ensure that the GameObject containing your MonoBehaviour is associated with a [SpatialOS entity]({{urlRoot}}/content/glossary#spatialos-entity). You can verify this by examining whether a `SpatialOSComponent` MonoBehaviour was added to your GameObject when your game is running
```

  * Ensure that the [requirements]({{urlRoot}}/content/gameobject/interact-spatialos-monobehaviours) for all your fields in your MonoBehaviour are satisfied. You can verify this by examining the [SpatialOS entity]({{urlRoot}}/content/glossary#spatialos-entity) associated with your GameObject in the [SpatialOS Inspector]({{urlRoot}}/content/glossary#inspector). In the SpatialOS Inspector, ensure that all relevant SpatialOS components are attached to your SpatialOS entity and that read and write access is delegated correctly for your worker.

<br/>

> #### <li> <b> When opening a Unity GDK solution in Visial Studio 2017, the file has no matching compilation unit </b>

Reviewer: ```suggestion
#### <li> <b> When opening a Unity GDK solution in Visual Studio 2017, the file has no matching compilation unit </b>
```
